src/components/ViewRequest/Details.tsx

```diff
@@ -17,6 +29,20 @@ const ViewRequestDetails = ({
 }) => {
   const params = useParams();
   const request = useRequest(Number(params.requestId));
+  const { isHRL } = useMyRoles();
+  const claimRequest = useClaimRequest();
+
```

Comment:
A thought to consider... move isHRL, restrictedStage and isHRLClaimable into the HRL API.  Then pass in the request data to the useClaimRequest hook, and return the isHRLClaimabe, and claimRequest functions.  That way the logic for HRL stays all within the API file itself.

Reply:
also, you can then remove the _spPageContextInfo from this file...